src/service/access.js

```diff
@@ -2,17 +2,44 @@ var Errors = require("../error/errors");
 var db = require("randoDB");
 var logger = require("../log/logger");
 var config = require("config");
+var userService = require("./userService");
+
+function sendUnauthorized (res) {
+  var response = Errors.toResponse(Errors.Unauthorized());
+  res.status(response.status).send(response);
+}
+
+function sendForbidden(res, ban) {
+  var response = Errors.toResponse(Errors.Forbidden(ban));
+  res.status(response.status).send(response);
+}
+
+function updateIp (user, ip) {
+  if (!user.ip || (user.ip && user.ip != ip)) {
+    logger.debug("[access.updateIp] Update to new ip[", ip ,"] for user: ", user.email);
+    user.ip = ip;
+  }
+}
 
 function checkAccess (req, res, next) {
-  var ip = req.ip;
-  var token = req.query.token;
-  logger.debug("[access.checkAccess] start. token: ", token, " ip: ", ip);
-  if (!token) {
-    logger.debug("[access.checkAccess] No Token. Send Unauthorized");
+  var ip = req.headers["x-forwarded-for"] || req.connection.remoteAddress;
+
+  var authHeader = req.headers.authorization;
```

Comment:
We use this method and `var firebaseInstanceId = req.get("FirebaseInstanceId");` in the same time. We should be consistent.